src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -311,5 +345,32 @@ private Uri CreateFullEndpointUri(Uri endPoint)
                 }
             }
         }
+
+        private IDictionary<string, string> ParseChallengeData(HttpResponse response)
+        {
+            IDictionary<string, string> data = new Dictionary<string, string>();
```

Comment:
Dictionaries are case senstive by default, so when you call thinkgs like `ContainsKey` it behaves case-sensitive, which is not what you want. Case insensitive dictionaries - https://stackoverflow.com/questions/13988643/case-insensitive-dictionary-with-string-key-type-in-c-sharp

Reply:
Surprisingly, according to the spec the values for PKeyAuth are actually case sensitive so this is the correct comparison.

Reply:
In that case, it would be good to add a comment about this, as it is certain nobody is going to remember this spec detail :)